pages/bills/[id]/index.js

```diff
@@ -1,39 +1,71 @@
 import BillDetails from "@/components/BillDetails";
-import Link from "next/link";
 import { useRouter } from "next/router";
 import useSWR from "swr";
 import styled from "styled-components";
+import { LinkButton, GhostLinkButton, GhostButton } from "@/components/Button";
+
+function fetcher(url) {
+  return fetch(url).then((response) => response.json());
+}
 
 export default function BillPageDetails() {
   const router = useRouter();
   const { id } = router.query;
 
-  const { data: bill, error, isLoading } = useSWR(`/api/bills/${id}`);
+  const { data: bill, error, isLoading } = useSWR(`/api/bills/${id}`, fetcher);
 
   if (isLoading || error) return <h2>Loading Bill..</h2>;
 
+  async function deleteBill() {
+    await fetch(`/api/bills/${id}`, {
+      method: "DELETE",
+    });
+    router.push("/");
+  }
+
   return (
-    <BillDetails title={bill.title} amount={bill.amount}>
-      {bill.participants.map((participant, index) => (
-        <li key={index}>
-          {participant.name} paid {participant.paid} $, and{" "}
-          {participant.balance}.
-        </li>
-      ))}
-      <Button href={`/bills/${id}/edit`}>Edit Bill</Button>
-      <Button href={"/"}>Back to Dashboard</Button>
-    </BillDetails>
+    <Wrapper>
+      <BillDetails title={bill.title} amount={bill.amount}>
+        {bill.participants.map((participant, index) => (
```

Comment:
Using index as a key in a list is generally not considered a good practice, as it can lead to issues with the React reconciliation algorithm when the order of the list items changes. for example when you delete the first item the index changes for all. It is recommended to use a unique identifier for each item instead. In this case, it would be better to use something like nano id or uid.